traffic_portal/app/src/common/modules/form/server/form.server.tpl.html

```diff
@@ -43,285 +42,245 @@
-                    <span ng-show="hasError(serverForm.routerHostName)" class="form-control-feedback"><i class="fa fa-times"></i></span>
-                </div>
+
+            <label ng-class="{'has-error': hasError(serverForm.routerHostName)}">Router Hostname</label>
+            <div ng-class="{'has-error': hasError(serverForm.routerHostName), 'has-feedback': hasError(serverForm.routerHostName)}">
+                <input name="routerHostName" type="text" class="form-control" ng-model="server.routerHostName" maxlength="256" pattern="\S*"/>
+                <small class="input-error" ng-show="hasPropertyError(serverForm.routerHostName, 'maxlength')">Too Long</small>
+                <small class="input-error" ng-show="hasPropertyError(serverForm.routerHostName, 'pattern')">No Spaces</small>
+                <span ng-show="hasError(serverForm.routerHostName)" class="form-control-feedback"><i class="fa fa-times"></i></span>
             </div>
-            <div class="form-group" ng-class="{'has-error': hasError(serverForm.routerPortName), 'has-feedback': hasError(serverForm.routerPortName)}">
-                <label class="control-label col-md-2 col-sm-2 col-xs-12">Router Port Name</label>
-                <div class="col-md-10 col-sm-10 col-xs-12">
-                    <input name="routerPortName" type="text" class="form-control" ng-model="server.routerPortName" ng-maxlength="256" autofocus>
-                    <small class="input-error" ng-show="hasPropertyError(serverForm.routerPortName, 'maxlength')">Too Long</small>
-                    <span ng-show="hasError(serverForm.routerPortName)" class="form-control-feedback"><i class="fa fa-times"></i></span>
-                </div>
+
+            <label ng-class="{'has-error': hasError(serverForm.routerPortName)}">Router Port Name</label>
+            <div ng-class="{'has-error': hasError(serverForm.routerPortName), 'has-feedback': hasError(serverForm.routerPortName)}">
+                <input name="routerPortName" type="text" class="form-control" ng-model="server.routerPortName" maxlength="256"/>
+                <small class="input-error" ng-show="hasPropertyError(serverForm.routerPortName, 'maxlength')">Too Long</small>
+                <span ng-show="hasError(serverForm.routerPortName)" class="form-control-feedback"><i class="fa fa-times"></i></span>
             </div>
-            <div class="modal-footer">
+
+            <fieldset>
+                <legend>Interfaces<button class="btn btn-success right-button" type="button" title="add a new interface" ng-click="addInterface()">Add Interface</button></legend>
+                <fieldset ng-repeat="inf in server.interfaces">
+                    <legend>
+                        <input my-directive type="text" ng-model="inf.name" required aria-label="Interface name" id="{{inf.name}}-name" name="{{inf.name}}-name" ng-class="{'has-error': hasError(serverForm[inf.name+'-name'])}" placeholder="eth0"/>
```

Comment:
my-directive?

Reply:
this text input acts weird for me. does it for you? i can only put my cursor at the start of the text.

Reply:
idk what "my-directive" is or why it's in there. I'll get rid of it.
as far as "acting weird," no, I don't think I have, but I'll look at it.

Reply:
directive is gone, can't reproduce not being allowed to put your cursor wherever you want

Reply:
oh, i understand what is going on

![image](https://user-images.githubusercontent.com/251272/85420763-e13ab480-b530-11ea-9feb-b7e44014bcc3.png)

eth0 is a placeholder. it looks like regular text so i was trying to double click it to select it all to replace it...

Reply:
maybe the placeholder was a bad idea because now it looks like text and the `update` or `create` button is disabled unless i change it....but if i want it to be `eth0` it looks like it's already filled in. maybe a better placeholder would be `interface-name`?

Reply:
That could work.